### ISSUE TITLE
Config Permalinks for page and post, and page_url

### ISSUE THREAD

**Issue by ickc** [jekyll/jekyll#4519]
In the documentation ([Permalinks](http://jekyllrb.com/docs/permalinks/)), it seems to suggest that in the config, only the posts' permalink can be configured site-wide. But when I test it, changing the permalink in config also affect pages. e.g. A page named `/categories.html` will becomes

| links | with config |
| --- | --- |
|`/categories/index.html` | `permalink: /:categories/:title/` |
| `/categories.html` | `permalink: /:categories/:title.html` |

Is it intentional?

Secondly, are there any way to configure the permalink site-wide in the config for pages and posts individually?

Lastly, this one is about getting the url of a post/page. From the documentation ([Templates](http://jekyllrb.com/docs/templates/)), there seems to be `post_url` but not `page_url` or something similar. So what's the best practice to link to a page? I'm asking this because if I change the permalink setting from `/:categories/:title/` to `/:categories/:title.html`, I need to manually change all the links to the pages, even if the source page location hasn't change. Are there existing way to handle this? (e.g. using capture??)

In short,

1. Can the 2 documentations mentioned above be clarified?
2. Can Jekyll added a way to config the permalink for post and page individually?
3. Can Jekyll also added `page_url` in addition to `post_url`?

Please correct me if I am misunderstand on anything and let me know if there's already existing way to accomplish / get around with the problems I mentioned.

Thanks.

**Comment by DirtyF** [comment]
Hi @ickc, can you share what are you trying to achieve?

Two things to be aware of:

1. You can access the list of all pages of your site through `site.pages` variable and link to pages URLs like it's done in the default jekyll template for instance: 

```
{% for my_page in site.pages %}
          {% if my_page.title %}
          <a href="{{ my_page.url | prepend: site.baseurl }}">{{ my_page.title }}</a>
          {% endif %}
{% endfor %}
```

2. You can define a custom permalink in the front matter of your page or post.

Hope it helps.

**Comment by ickc** [comment]
In a sense what I want to do is for a theme to work in different sites, where each of them might have different config. E.g. In the nav and sidebar, there are links to some pages like home, archive, about, authors, categories, tags, etc.

Now given the theme, if the user choose to use a different config, and incidentally changed the permalink from say `/:categories/:title/` to `/:categories/:title.html`, then all links to pages in the nav and sidebar will be broken.

So what I want is a way to refer to the links to the pages given the links in the source, but independent of the output site. I.e. the `page_url` I was talking about.

**Comment by parkr** [comment]
> In the documentation (Permalinks), it seems to suggest that in the config, only the posts' permalink can be configured site-wide.

This should be updated. You're right; in v3, this does affect pages.

For posts, it's specify the permalink via the `collections.posts`:

```yaml
collections:
  posts:
    output: true
    permalink: /:year/:month/:date/:title/
```

**Comment by svivian** [comment]
> From the documentation (Templates), there seems to be post_url but not page_url or something similar.

+1 for this, so annoying that there is no 'stable' way to link to pages. Looping through `site.pages` might work for navigation, but it's infeasible for links in your content.

**Comment by shaungrady** [comment]
+1 for a `page_url`

**Comment by zapjelly** [comment]
also +1 for a page_url

**Comment by parkr** [comment]
The new `{% link %}` tag in v3.2 will fix this issue. Thanks! See #4624 for details.

**Comment by ickc** [comment]
Thanks!
